src/NCronJob/Scheduler/TaskScheduler/DeterministicTaskScheduler.cs

```diff
@@ -0,0 +1,45 @@
+namespace NCronJob;
+
+/// <summary>
+/// Deterministic and ordered execution: Use when the order of task execution is critical, and tasks may have dependencies that require strict ordering.
+/// </summary>
+internal class DeterministicTaskScheduler : TaskScheduler
+{
+    private readonly LinkedList<Task> tasks = [];
+    private readonly object @lock = new();
```

Comment:
With .NET 9 we can do this:

```csharp
#if NET8_0
    private readonly object @lock = new();
#else
    private readonly Lock @lock = new();
```

Reply:
This change required opting into the preview features so I added the following to the csproj file:
```xml
  <PropertyGroup>
    <EnablePreviewFeatures>true</EnablePreviewFeatures>
  </PropertyGroup>
```

Reply:
Ahhh I thought they put this from preview to stable. Okay, maybe we should put this up separately or, alternatively, remove this if it doesn't get stable until `net9.0`.

Reply:
It's up to you. I removed it from `DeterministicTaskScheduler` as I'm now just using ConcurrentQueue for thread management. `ObservablePriorityQueue` is currently the only data structure where we need it.

Reply:
Let's keep it - my hope is that it gets stable. We can switch it prior to .net 9 GA.